src/app/container/deregister-modal/deregister-modal.component.ts

```diff
@@ -15,20 +15,44 @@
  */
 
 import { Component, Input, OnInit } from '@angular/core';
+import { takeUntil } from 'rxjs/operators';
+import { ConfirmationDialogData } from '../../confirmation-dialog/confirmation-dialog.component';
+import { ConfirmationDialogService } from '../../confirmation-dialog/confirmation-dialog.service';
+import { Base } from '../../shared/base';
+import { bootstrap4mediumModalSize } from '../../shared/constants';
 import { RegisterToolService } from './../register-tool/register-tool.service';
 
 @Component({
   selector: 'app-deregister-modal',
   templateUrl: './deregister-modal.component.html',
   styleUrls: ['./deregister-modal.component.css']
 })
-export class ModalComponent implements OnInit {
+export class ModalComponent extends Base implements OnInit {
   @Input() refreshMessage;
 
-  constructor(private registerToolService: RegisterToolService) {}
+  constructor(private registerToolService: RegisterToolService, private confirmationDialogService: ConfirmationDialogService) {
+    super();
+  }
 
   ngOnInit() {}
   deregister() {
     this.registerToolService.deregisterTool();
   }
+
+  openDeleteDialog() {
+    const confirmationDialogData: ConfirmationDialogData = {
+      title: 'Are you sure you wish to delete this tool?',
+      message: `All information associated with this tool will be deleted.`,
+      cancelButtonText: 'Close',
```

Comment:
It was like this, but I think this should say `Cancel`. Makes it more explicit that you aren't deleting the tool, whereas `Close`, I have to think about it a little bit.

Reply:
Done.